libpod/container_internal_linux.go

```diff
@@ -1458,11 +1458,38 @@ func (c *Container) generateCurrentUserPasswdEntry() (string, error) {
 	if uid == 0 {
 		return "", nil
 	}
+
 	u, err := user.LookupId(strconv.Itoa(rootless.GetRootlessUID()))
 	if err != nil {
 		return "", errors.Wrapf(err, "failed to get current user")
 	}
-	return fmt.Sprintf("%s:x:%s:%s:%s:%s:/bin/sh\n", u.Username, u.Uid, u.Gid, u.Username, c.WorkingDir()), nil
+
+	// Lookup the user to see if it exists in the container image.
+	_, err = lookup.GetUser(c.state.Mountpoint, u.Username)
```

Comment:
Corner case: would it make sense to do lookup by UID instead of name? This is super-unlikely, but if the podman user has a UID already defined in the image's `/etc/passwd`, appending a new entry is going to be mostly a NOP: `ls` and `ps` and anything that does lookups is going to show the preexisting username.

Reply:
I considered that, and decided it's probably not worth the potential performance hit of a duplicate lookup.